src/store/reducers/UserSlice.ts

```diff
@@ -41,8 +52,20 @@ export const counterSlice = createSlice({
     deleteTodo: (state, action: PayloadAction<string>) => {
       state.todos = state.todos.filter((todo) => todo.id !== action.payload);
     },
+    updateTodo: (state, action: PayloadAction<TodoData>) => {
+      state.todos = state.todos.map((todo) =>
+        todo.id === action.payload.idd
+          ? {
+              ...todo,
+              message: action.payload.message,
```

Comment:
Use destructuring, and it will look like {...todo, message, date, dateExperation: expDate}

Reply:
Add